preprocessors/openstreetmap/osm_service.py

```diff
@@ -372,135 +326,97 @@ def compute_new_node(node_params, bearing, bbox_coordinates):
-            # so compute for latitude while set longitude to
-            # lon_min.
-            x2 = radians(lon_min)
-            y2 = ((x2 - lon1) * math.cos(lat1) / math.tan(theta))
-            lat2 = degrees(y2 + lat1)
-            lon2 = degrees(x2)
-    elif bearing >= 0 or bearing == 360:  # The street intercepts only
-        # at the top side of the bounds, so compute for longitude while
-        # setting the latitude to lat_max.
-        y2 = radians(lat_max)
-        x2 = ((math.tan(theta) * (y2 - lat1)) / math.cos(lat1))
-        y2 = degrees(y2)
-        x2 = degrees(x2 + lon1)
-        lat2 = y2
-        lon2 = x2
-    elif bearing == 45:  # The street intercepts only
-        # at the NE edge of the bounds.
+    # The bearing indicates the side of the bounding box that intercepts the
+    # street.
+    flag = False
+    if bearing > 0 and bearing < 90:  # if true,  the intercept is at
+        # the top or the right side.
+        flag = True
+        lat2 = radians(lat_max)
+        lat2, lon2 = get_new_node_coordinates(flag, lat1, lon1, lat2, theta)
+        # If validation returns true, the intercept is at the top side,
+        validated = validate_new_node_coordinates(lat2, lon2, bbox_coordinates)
+        if not validated:  # if true,
+            # the intercept takes place at the right side.
+            flag = False
+            lon2 = radians(lon_max)
+            lat2, lon2 = get_new_node_coordinates(
+                flag, lat1, lon1, lon2, theta)
+    elif bearing > 90 and bearing < 180:  # if true,
+        # the intercept is at the bottom or the right side.
+        flag = True
+        lat2 = radians(lat_min)
+        lat2, lon2 = get_new_node_coordinates(flag, lat1, lon1, lat2, theta)
+        # If validation returns true, the intercept is at the bottom side
+        validated = validate_new_node_coordinates(lat2, lon2, bbox_coordinates)
+        if not validated:  # if true,
+            # the intercept takes place at the right side.
+            flag = False
+            lon2 = radians(lon_max)
+            lat2, lon2 = get_new_node_coordinates(
+                flag, lat1, lon1, lon2, theta)
+    elif bearing > 180 and bearing < 270:  # if true,
+        # the intercept is at the bottom or the left side.
+        flag = True
+        lat2 = radians(lat_min)
+        lat2, lon2 = get_new_node_coordinates(flag, lat1, lon1, lat2, theta)
+        # If validation returns true, the intercept is at the bottom side
+        validated = validate_new_node_coordinates(lat2, lon2, bbox_coordinates)
+        if not validated:  # if true,
+            # the intercept takes place at the left side.
+            flag = False
+            lon2 = radians(lon_min)
+            lat2, lon2 = get_new_node_coordinates(
+                flag, lat1, lon1, lon2, theta)
+    elif bearing > 270 and bearing < 360:  # if true,
+        # the intercept is at the top or the left side.
+        flag = True
+        lat2 = radians(lat_max)
```

Comment:
What is going on with all these conversions back and forth between degrees and radians?

Reply:
Want coordinates to be returned in degrees. Meanwhile Python trigonometric functions work with radians. It's however being handled better now.